index.html

```diff
@@ -1145,14 +1444,118 @@ <h2>
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |origin| be |baseURL|'s [=url/origin=].
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |fallbackTimestamp| be <a href=
+                  "https://github.com/w3c/hr-time/issues/165">w3c/hr-time #165</a>.
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |declarativeResult| be the result of running the [=/declarative push message
+                  parser=] given |bytes|, |origin|, |baseURL|, and |fallbackTimestamp|.
+                </p>
+              </li>
+              <li>
+                <p>
+                  If |declarativeResult| is not failure:
+                </p>
+                <ol>
+                  <li>
+                    <p>
+                      Let |notification| be |declarativeResult|'s [=declarative push message parser
+                      result/notification=].
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      Set |notification|'s [=notification/service worker registration=] to
+                      |registration|.
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      If |declarativeResult|'s [=declarative push message parser result/mutable
+                      notification=] is true:
+                    </p>
+                    <ol>
+                      <li>
+                        <p>
+                          [=/Fire a functional event=] named "`pushnotification`" using
+                          {{PushNotificationEvent}} on |registration| with the following
+                          properties:
+                        </p>
+                        <dl>
+                          <dt>
+                            `notification`
+                          </dt>
+                          <dd>
+                            |notification|
+                          </dd>
+                          <dt>
+                            `appBadge`
+                          </dt>
+                          <dd>
+                            |declarativeResult|'s [=declarative push message parser result/app
+                            badge=]
+                          </dd>
+                        </dl>
+                      </li>
+                      <li>
+                        <p>
+                          Wait for all of the promises in the [=ExtendableEvent/extend lifetime
+                          promises=] of the dispatched event to resolve.
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          If a notification has been shown through {{showNotification()}} at this
+                          point, then abort these steps.
```

Comment:
I mean, this statement is monkey patching that. Certainly seems reasonable to tidy it up though. (`preventDefault()` wouldn't guarantee that a notification is shown.)

Reply:
Once https://github.com/whatwg/notifications/pull/227 lands this can be checked here instead of `showNotification()`. Does that seem reasonable? Anything else remaining?